cpp/cograph_recognition/cograph/Recognition.cpp

```diff
@@ -0,0 +1,233 @@
+#include <recognition/CographRecognition.hpp>
+#include <list>
+#include <set>
+
+#include <graph/GraphTools.hpp>
+namespace Koala {
+enum class Type{
+ZEROONE,
+VERTEX
+};
+enum class Marked{
+    UNMARKED,
+    MARKED,
+    MARKED_AND_UNMARKED
+}
+class CoNode{
+    private:
+    Type type;
+    int number;
+    Marked marked;
+    int md, d;
+    //md is the current number of children, which have been both "marked" and "unmarked"
+    //d is the current number of children
+
+    CoNode* head_of_list_of_children;
+    CoNode *next, *prev;//in list of children of its parent
+    CoNode *parent;
```

Comment:
I'd suggest std::list<CoNode> children and std::list<CoNode>::iterator next, previous